pkg/scheduler/scheduler.go

```diff
@@ -121,6 +144,21 @@ func (s *Scheduler) schedule(ctx context.Context) {
 			e.inadmissibleMsg = "cohort used in this cycle"
 			continue
 		}
+
+		if s.waitForPodsReady {
+			if !s.cache.PodsReadyForAllAdmittedWorkloads(ctx) {
+				log.V(5).Info("Waiting for all admitted workloads to be in the PodsReady condition")
+				// Block admission until all currently admitted workloads are in
+				// PodsReady condition if the waitForPodsReady is enabled
+				if err := workload.UpdateStatus(ctx, s.client, e.Obj, kueue.WorkloadAdmitted, metav1.ConditionFalse, "Waiting", "waiting for all admitted workloads to be in PodsReady condition"); err != nil {
+					log.Error(err, "Could not update Workload status")
+					continue
```

Comment:
why `continue` here? failing to update the status here isn't critical, we shouldn't give up on admitting the workload here.

Reply:
My reasoning was that if the update fails it is likely due to conflicts as `e` holds an outdated workload state, so the actual admission is also likely to fail.

However, we need to be prepared anyway to handle outdated workload state during admission anyway, so we should be ok with trying admission (and it may actually succeed.  Reverting the `continue`.